### ISSUE TITLE
Releasing the GIL

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/ucx-py#251]
Currently we don't release the GIL in the code. Perhaps this is something we should? If so, it would be good to identify where we can release the GIL. If this is not something we should do, it would be good to understand why.

**Comment by jakirkham** [comment]
@madsbk, do you have any thoughts on this?

**Comment by madsbk** [comment]
I think we should be able to release the GIL at most UCX API calls but they are all non-blocking so I don't think it will have a significant impact on the performance.

**Comment by mrocklin** [comment]
Agreed.  I would wait on this until profiling showed that it was important.

**Comment by jakirkham** [comment]
I'm not as concerned about performance speedup as handling major issues like hangs at this stage. The latter hurts usability so is worth thinking about IMHO.

**Comment by mrocklin** [comment]
I agree that hangs are bad.  Are you saying that holding onto the GIL is
causing UCX to hang?

**Comment by jakirkham** [comment]
Wanted to check and see if there would be any problem pursuing this path should it be needed. Am hearing there shouldn’t be which is good.

**Comment by mrocklin** [comment]
In my mind it might be helpful, but really we should focus on getting a
benchmark up and running quickly so that we can make these kinds of
determinations.  I think that once someone is actually playing with
dask-cudf joins then we'll know a ton more about what work needs to be
done.  I'm personally less excited about cleanup work until we have a
benchmark to help give direction and priorities.

**Comment by pentschev** [comment]
We have since made several improvements to GIL release, plus UCXX should mostly cover this correctly. Therefore, it shouldn't be relevant anymore, closing.
